www/js/lib/uiUtil.js

```diff
@@ -43,34 +43,87 @@ define(rqDef, function(settingsStore) {
+                if (callback) callback(); // Calling back as soon as possible speeds up extraction
+                if (params.useCanvasElements) {
+                    // Replace images by canvas. This is necessary for some browsers with obsolete anti-fingerprinting protection, like IceCat 60.7
+                    webpMachine.constructor.replaceImageWithCanvas(node, canvas);
+                } else {
+                    // For standard browsers that have access to the canvas, we simply convert the canvas to a data URI
+                    node.setAttribute(nodeAttribute, canvas.toDataURL());
+                }
             }).catch(function (err) {
                 console.error('There was an error decoding image in WebpMachine', err);
                 if (callback) callback();
             });
         } else {
-            var blob = new Blob([content], {
-                type: mimeType
-            });
-            var url = URL.createObjectURL(blob);
-            if (callback) callback(url);
-            node.addEventListener('load', function () {
-                URL.revokeObjectURL(url);
-            });
-            node.setAttribute(nodeAttribute, url);
+            if (callback) callback(); // Calling back as soon as possible speeds up extraction
+            // In browsers that support WebP natively, or for non-WebP images, we can simply convert the Uint8Array to a data URI
+            // DEV: we use FileReader method because btoa fails on utf8 strings (in SVGs, for example)
+            // See https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem
+            // This native browser method is very fast: see https://stackoverflow.com/a/66046176/9727685
+            var myReader = new FileReader();
+            myReader.onloadend = function () {
+                var url = myReader.result;
+                node.setAttribute(nodeAttribute, url);
+            };
+            myReader.readAsDataURL(new Blob([content], { type: mimeType }));
+
+            // Below is BLOB URL-based method; slower and requires callback on every image load to revoke the BLOB
+            // var imgBlob = new Blob([content], {
+            //     type: mimeType
+            // });
+            // var url = URL.createObjectURL(imgBlob);
+            // if (callback) callback(url);
+            // node.addEventListener('load', function () {
+            //     URL.revokeObjectURL(url);
+            // });
+            // node.setAttribute(nodeAttribute, url);
+        }
+    }
+
+    /**
+     * Determines whether the Canvas Elements Workaround for decoding WebP images is needed, and sets UI accordingly
+     * Note that the workaround will never be used in Service Worker mode, but we still need to determine it in case the user switches to jQuery mode
+     * @returns {Boolean} A value to indicate whether the workaround may needed 
+     */
+    function determineCanvasElementsWorkaround() {
```

Comment:
I find it difficult to understand this function, because it handles many things at the same time.
Maybe it could be split between handling the UI (only showing/hiding the div) and setting the param/settingsStore

Reply:
I've greatly simplified this function now and added clear comments for the things it does. The structure was confused before. Let me know if you think it still needs splitting in two.

Reply:
It's more clear, but the header should mention it also sets the `params`.

Reply:
Note added to header.